tools/c7n_oci/c7n_oci/resources/object_storage.py

```diff
@@ -0,0 +1,150 @@
+# Copyright The Cloud Custodian Authors.
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+
+import oci.object_storage
+
+from c7n.utils import type_schema
+from c7n_oci.actions.base import OCIBaseAction, RemoveTagBaseAction
+from c7n_oci.provider import resources
+from c7n_oci.query import QueryResourceManager
+
+log = logging.getLogger('custodian.oci.resources.object_storage')
+
+@resources.register('bucket')
+class Bucket(QueryResourceManager):
+    """Oracle Cloud Infrastructure Bucket Resource
+
+    :example:
+
+    Returns all Bucket resources in the tenancy
+
+    .. code-block:: yaml
+
+        policies:
+            - name: find-all-bucket-resources
+              resource: oci.bucket
+
+    """
+
+    class resource_type:
+        doc_groups = ['ObjectStorage']
+        service = 'oci.object_storage'
+        client = 'ObjectStorageClient'
+        enum_spec = ('list_buckets', 'items[]', None)
+        extra_params = {'compartment_id','namespace_name'}
+        resource_type = 'OCI.ObjectStorage/Bucket'
+        id = 'identifier'
+        name = "display_name"
+        search_resource_type = 'bucket'
+
+
+    def _construct_query_params(self):
```

Comment:
this method should be moved to the base class, it seems to be copy and pasted into all the resource managers.